dev/planning-example.py

```diff
@@ -0,0 +1,265 @@
+at that location.
+"""
+# Define an observer by full specification
+#
+# Because there may be many attributes for an observer, I suggest using
+# keyword parameters for most of them.
+from astroplan.entity import Observer
+
+obs = Observer(name='Subaru Telescope',
+               longitude='-155:28:48.900',
+               latitude='+19:49:42.600',
+               elevation=4163,
+               pressure=615,
+               relative_humidity=0.11,
+               temperature=0,
+               timezone='US/Hawaii',
+               description="Subaru Telescope on Mauna Kea, Hawaii")
+
+# longitude and latitude would be represented internally by astropy
+# angles.Longitude/Latitude.  So any value that can naturally initialize
+# one of those objects could be passed as the parameter to Observer
+# Other items are represented internally using the appropriate astropy
+# Quantities and Units and so can be initialized similarly.
+
+# It's also possible to initialize using astropy objects and other
+# direct entities
+from astropy.coordinates.angles import Longitude, Latitude
+import astropy.units as u
+import pytz
+
+obs = entity.Observer(name='Keck 1',
+                      longitude=Longitude(155.4750 * u.degree),
+                      latitude=Latitude(19.8264 * u.degree)
+                      elevation=4160 * u.m,
+                      pressure=615 * u.hPa,
+                      relative_humidity=0.11,
+                      temperature=0.0 * u.C,
+                      timezone=pytz.timezone('US/Hawaii'),
+                      description="W.M. Keck 1 Telescope on Mauna Kea, Hawaii")
+
+# It would also be desirable to be able to have a small database of
+# common telescopes.  Maybe this can at first simply take the form of
+# a python module:
+from astroplan import sites
+
+obs = sites.get_observer('keck1')
+
+# Environmental conditions should be updatable.  Suggest using keyword
+# parameters (this would make it easy to update an internal AltAz coordinate):
+#
+obs.set_env_conditions(pressure=600 * u.hPa, relative_humidity=0.2,
+                       temperature=10 * u.C)
+
+# It's easy to construct date/times from the observer, since it knows about
+# its timezone.  These dates can be passed directly to target check methods.
+
+# dates default to midnight if not otherwise specified
+day = obs.get_date("2015-05-01")
+
+# Can get more and more specific
+t1 = obs.get_date("2015-05-01 21:00")
+t1 = obs.get_date("2015-05-01 21:33:41")
+
+# default is current time
+cur_time = obs.get_date()
+
+# Dates are assumed to be in the observer's time zone unless a time zone
+# specifier is given
+t2_utc = obs.get_date("2015-05-01 21:33:41 UTC")
+
+"""
+We can ask about typical times of interest at this observing position
+Returned dates are assumed to be in the observer's time zone for convenience.
+The date parameter, if given, can be an astropy Time or a datetime instance.
+If no parameter is given, the current date/time is assumed.
```

Comment:
A subtle question about ``date``: it's not obvious what "date" that's referring to.  That is, are we asking about the dark period that's the *morning* of that day, or the *night* of that day?  

I moderately common (but not universal) convention in astronomy is to say a date and implictly take that to mean "the night of that date".  If we take that convention, then instead of ``date`` the keyword should be ``nightof`` or similar.  However, that has a problem: what if I'm in pacific time and specify the date as '2015-05-01 01:00:00 Pacific'?  A human reading that (or an equivalent datetime or Time object) would probably assume that means the *morning* of May 1/night of April 30, but the ``nightof`` convention would take it to be the night of May 1/May 2.

So an alternative is to have it be "the dark period closest to that date". That would require some calculation, though, so it has a performance overhead.  One way to mitigate that might be to special-case "pure dates", like the string '2015-05-01' or a datetime.date object.

Another alternative is to *only* allow pure dates, which could go with the "night of" convention.

There might be other options, but those are the ones that come to mind for me.